src/software/amazon/ion/Timestamp.java

```diff
@@ -83,6 +83,18 @@
     private static final int NO_SECONDS = 0;
     private static final BigDecimal NO_FRACTIONAL_SECONDS = null;
 
+    // 0001-01-01T00:00:00.0Z in millis
+    private static final long MINIMUM_ALLOWED_TIMESTAMP_IN_MILLIS = -62135769600000L;
+
+    // 0001-01-01T00:00:00.0Z in millis
+    static final BigDecimal MINIMUM_ALLOWED_FRACTIONAL_MILLIS = new BigDecimal(MINIMUM_ALLOWED_TIMESTAMP_IN_MILLIS);
+
+    // 9999-12-31T23:59:59.999999999999Z in millis
+    static final BigDecimal MAXIMUM_ALLOWED_FRACTIONAL_MILLIS = new BigDecimal("253402300799999.999999999");
```

Comment:
This isn't correct: I can always add another 9 to get a larger, but still valid, input.   By design, this library does not place bounds on precision, because its goal is to faithfully represent the Ion data model.

The correct upper-bound check is, effectively, `< 10000T`.

Reply:
good point, will fix that